artichoke-backend/src/extn/core/env/env.rb

```diff
@@ -36,10 +36,58 @@ def delete_if
     to_h
   end
 
+  def each
+    return to_enum(:each) unless block_given?
+
+    to_h.each { |key, value| yield key, value }
+  end
+
+  def each_key
+    return to_enum(:each_key) unless block_given?
+
+    to_h.each_key { |key| yield key }
+  end
+
+  def each_pair
+    return to_enum(:each) unless block_given?
+
+    to_h.each_pair { |key, value| yield key, value }
+  end
+
+  def each_value
+    return to_enum(:each_value) unless block_given?
+
+    to_h.each_value { |value| yield value }
+  end
+
   def empty?
     to_h.empty?
   end
 
+  def fetch(name, default = nil)
+    value = self[name]
+
+    return value unless value.nil?
+
+    return yield name if block_given?
+
+    return default unless default.nil?
+
+    raise KeyError, 'Variable ' + name + " doesn't exist"
```

Comment:
can we convert this to use `String` interpolation?

It also appears MRI uses a different error message: `"key not found: #{name}"`

```
[2.6.3] > ENV.fetch('aaaaaaaa')
Traceback (most recent call last):
        6: from /usr/local/var/rbenv/versions/2.6.3/bin/irb:23:in `<main>'
        5: from /usr/local/var/rbenv/versions/2.6.3/bin/irb:23:in `load'
        4: from /usr/local/var/rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/irb-1.0.0/exe/irb:11:in `<top (required)>'
        3: from (irb):11
        2: from (irb):11:in `rescue in irb_binding'
        1: from (irb):11:in `fetch'
KeyError (key not found: "aaaaaaaa")
```

Reply:
Sure, I've fixed a remark